jit: Fix float-point instruction

qwe661234 opened the pull request:
1. Trace the file src/softfloat.h in gen-jit script to include an enumeration for floating point instructions.
2. To simplify the definition of JIT gencode, we adjust the data structure of riscv_internal by placing the fields needed by JIT at the top.

Solve: https://github.com/sysprog21/rv32emu/issues/169

jserv commented on src/softfloat.h:
```diff
@@ -11,19 +11,19 @@
 /* clang-format off */
 enum {
-    //                    ....xxxx....xxxx....xxxx....xxxx
+    /*                    ....xxxx....xxxx....xxxx....xxxx */
```
Don't do this. Minimize the necessary changes.

jserv replied:
Python script should be aware of C++ style comments.

jserv commented on src/riscv_private.h:
```diff
@@ -130,6 +116,21 @@ struct riscv_internal {
 
     bool compressed; /**< current instruction is compressed or not */
+
+#if RV32_HAS(GDBSTUB)
```
Why did you make such change?

qwe661234 replied:
If we place the fields needed by JIT at the top of the data structure, we could reduce the definition in gencode. Otherwise, we need to define a data structure for GDB in gencode.

jserv replied:
Explain in git commit messages.

jserv reviewed (changes requested):
Improve the git commit messages. ChatGPT can help refine.

jserv commented on src/compile.c:
```diff
@@ -208,6 +208,20 @@ RVOP(sh, {MEMORY_FUNC(uint16_t, 0)})
+#if RV32_HAS(EXT_F)
+RVOP(flw, {
+    GEN("  addr = rv->X[%u] + %u;\n", ir->rs1, ir->imm);
+    GEN("  rv->F_int[%u] = * (const uint32_t *) (m->mem_base + addr);\n",
```
Should be `*((const uint32_t *) (m->mem_base + addr));`.

jserv commented on src/compile.c:
```diff
@@ -208,6 +208,20 @@ RVOP(sh, {MEMORY_FUNC(uint16_t, 0)})
+/* FSW */
+RVOP(fsw, {
+    GEN("  addr = rv->X[%u] + %u;\n", ir->rs1, ir->imm);
+    GEN("  *(uint32_t *) (m->mem_base + addr) = rv->F_int[%u];\n", ir->rs2);
```
Ditto. Use `*((uint32_t *) (m->mem_base + addr))`.

qwe661234 committed:
jit: Fix float-point instruction

1. Trace the file src/softfloat.h in gen-jit script to include an enumeration
for floating point instructions.
2. To simplify the definition of JIT gencode, we adjust the data structure of riscv_internal
by placing the fields needed by JIT at the top.

Solve: #169
